subrpcserver_config.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+)
+
+// subRpcServerConfigs is special sub-config in the main configuration that
+// houses the configuration for the optional sub-servers. These sub-RPC servers
+// are meant to house experimental new features that may eventually make it
+// into the main RPC server that lnd exposes. Special methods are present on
+// this struct to allow the main RPC server to create and manipulate the
+// sub-RPC servers in a generalized manner.
+type subRpcServerConfigs struct {
```

Comment:
I think using a separate config file per sub server in the `.lnd` directory is simpler. No reflection needed and less references from lnd to the sub services. Are there serious downsides? I can even imagine a separate config to be more in line with the plan to unbundle lnd separate processes.